### ISSUE TITLE
unable to delete disk when using deleteByIdAsync with service callback

### ISSUE THREAD

**Issue by ashishdhar** [Azure/azure-sdk-for-java#1905]
I am facing this issue while using azure SDK 1.1.0 or higher in deleting a disk.
azure.disks().deleteByIdAsync(diskid, deleteHandler);
FAILED with java.util.NoSuchElementException: Sequence contains no elements
at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:115)
at rx.internal.operators.DeferredScalarSubscriber.complete(DeferredScalarSubscriber.java:83)
at rx.internal.operators.OnSubscribeTakeLastOne$TakeLastOneSubscriber.onCompleted(OnSubscribeTakeLastOne.java:55)

Has this been fixed because it seems related to the prior issue https://github.com/Azure/azure-sdk-for-java/issues/1648?

The below code works but i wanted to use the service call back one.
```
completebale c = azure.disks().deleteByIdAsync(diskid)
c.await();
```

**Comment by anuchandy** [comment]
@ashishdhar thanks for reporting this.

It seems, in general ServiceCallBack based deleteByIdAsync is broken. This [method](https://github.com/Azure/autorest-clientruntime-for-java/blob/master/client-runtime/src/main/java/com/microsoft/rest/ServiceFuture.java#L95) is used to convert Observable to ServiceCallBack, but the Observable input to this method is created from Completable [here](https://github.com/Azure/azure-sdk-for-java/blob/master/azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/arm/collection/implementation/CreatableWrappersImpl.java#L37), since the produced Observable from Completable is value-less, calling `last()` on it cause this error.

The runtime should be fixed to have an overload that takes Completable and convert it to ServiceCallback.

**Comment by martinsawicki** [comment]
fixed in v1.4
